release/create_release_archives.sh

```diff
@@ -15,16 +15,12 @@
 #
 ################################################################################
 
-set -o errexit
-set -o pipefail
-set -x
-
 # This script primarily exists for Cloud Builder.  This script
 # reads artifacts from a specified directory, generates tar files
 # based on those artifacts, and then stores the tar files
 # back to the directory.
 
-TEMP_DIR="$(mktemp -d /tmp/istio.version.XXXX)"
```

Comment:
why?

Reply:
mktemp will be performed always in the beginning. but `rm -rf "$TEMP_DIR"` in the end. sometimes, there is error in the middle so that the "$TEMP_DIR" cannot be removed as expected.

Reply:
I think we want to just a clean and new temp folder always to avoid confusion. Cleaning up temp files is typically best effort.

Reply:
see this example:
```
chunlins-mbp:istio clyang$ ./release/create_release_archives.sh 
++ mktemp -d /tmp/istio.version.XXXX
+ TEMP_DIR=/tmp/istio.version.wORR
+ BASE_DIR=/tmp/istio.version.wORR
+ ISTIOCTL_SUBDIR=istioctl
+ OUTPUT_PATH=
+ VER_STRING=
+ getopts d:i:o:v: arg
+ [[ -z /tmp/istio.version.wORR ]]
+ [[ -z '' ]]
+ usage
+ echo './release/create_release_archives.sh
    -d <path> path to use for temp directory                  (optional, randomized default is /tmp/istio.version.wORR )
    -o <path> path where build output/artifacts are stored    (required)
    -i <name> subdirectory in -o path to use for istioctl     (optional)
    -v <ver>  version info to include in filename (e.g., 1.0) (required)'
./release/create_release_archives.sh
    -d <path> path to use for temp directory                  (optional, randomized default is /tmp/istio.version.wORR )
    -o <path> path where build output/artifacts are stored    (required)
    -i <name> subdirectory in -o path to use for istioctl     (optional)
    -v <ver>  version info to include in filename (e.g., 1.0) (required)
+ exit 1
```
The temp folder is created but never be cleaned.

Reply:
You can add a trap <func> EXIT to make sure cleanup happens even in error path

Reply:
fixed